cirq/ops/pauli_string.py

```diff
@@ -179,6 +179,12 @@ def __str__(self):
 
         return prefix + '*'.join(factors)
 
+    def _unitary_(self) -> np.ndarray:
+        u = np.array([self.coefficient], dtype=np.complex128)
+        for pauli in self.values():
+            u = np.kron(u, protocols.unitary(pauli))
```

Comment:
If we're really being safe I would also add `_apply_unitary_`, to ensure a 20-entry pauli string does not cause a memory error.